File: examples/bench/fib.c
/* Hand-written C reference for the fib(40) benchmark.
 * This is the baseline MFL is measured against: the same naive,
 * doubly-recursive algorithm the compiler emits from fib.mfl.
 *
 *   cc -O2 -o fib_c examples/bench/fib.c
 *   ./fib_c
 */
#include <stdio.h>

static long fib(long n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main(void) {
    printf("%ld\n", fib(40));
    return 0;
}

File: examples/bench/fib.rs
// Rust reference for the fib(40) benchmark — naive doubly-recursive,
// matching the MFL and C implementations for an apples-to-apples compare.
//
//   rustc -O -o fib_rs examples/bench/fib.rs
//   ./fib_rs

fn fib(n: i64) -> i64 {
    if n < 2 { return n; }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    println!("{}", fib(40));
}
